### PR TITLE
refactor: Encapsulate lease timing in a separate nested class

### PR DESCRIPTION
**Pull request by jskeet**
Personally I think this makes everything significantly clearer, but YMMV :)

The auto-extend throttle calculation looks a bit odd to me - which is clearer when it's extracted from everything else. Please check that carefully.

### CODE REVIEW

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs [hunk 1]
```diff
@@ -73,14 +73,8 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
...
         // These values are validated in Settings.Validate() above, so no need to re-validate here.
-        _ackExtensionWindow = settings.AckExtensionWindow ?? DefaultAckExtensionWindow;
-        _ackDeadlineSeconds = (int) ((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
-        // For exactly-once delivery subscriptions, AckDeadline value should be the default value of 60 seconds or the user provided value.
-        _ackDeadlineSecondsForExactlyOnceDelivery = (int) ((settings.AckDeadline ?? DefaultAckDeadlineForExactlyOnceDelivery).TotalSeconds);
-        // Ensure the duration between lease extensions is at least MinimumLeaseExtensionDelay (5 seconds).
-        // The minimum allowable lease duration is 10 seconds, so this will always be reasonable.
-        _autoExtendDelay = GetAutoExtendDelay(_ackDeadlineSeconds);
-        _autoExtendDelayForExactlyOnceDelivery = GetAutoExtendDelay(_ackDeadlineSecondsForExactlyOnceDelivery);
+        _normalLeaseTiming = new LeaseTiming(settings.AckDeadline ?? DefaultAckDeadlineForExactlyOnceDelivery, settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
```
Comment (Rishabh-V) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs:76]: Typo - Just `DefaultAckDeadline` here.

**Review by Rishabh-V — APPROVED** [review]
This looks good Jon. Just one typo. Thanks for refactoring and making it cleaner 👍🏼

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs:76]: Gah, thank you. (Slightly worrying that this wasn't caught by tests, but hey...)

Comment (jskeet) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs]: Fixed, will merge when the new commit has gone green.

**Pull request merged**